modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java

```diff
@@ -0,0 +1,212 @@
+		}
+
+		List<String> supportedLanguages =
+			_supportedLanguageManager.getSupportedLanguages(
+				_deepLTranslatorConfiguration);
+
+		String targetLanguageCode = _getLanguageCode(
+			translatorPacket.getTargetLanguageId());
+
+		if (!_verifyLanguage(supportedLanguages, targetLanguageCode)) {
+			_log.error(
+				StringBundler.concat(
+					"No target language available for ", targetLanguageCode,
+					". Supported languages are: ",
+					StringUtil.merge(
+						supportedLanguages, StringPool.COMMA_AND_SPACE)));
+
+			return translatorPacket;
+		}
+
+		String sourceLanguageCode = _getLanguageCode(
+			translatorPacket.getSourceLanguageId());
+
+		Map<String, String> translatedFieldsMap = new HashMap<>();
+
+		Map<String, String> fieldsMap = translatorPacket.getFieldsMap();
+
+		for (Map.Entry<String, String> entry : fieldsMap.entrySet()) {
+			translatedFieldsMap.put(
+				entry.getKey(),
+				_translate(
+					_deepLTranslatorConfiguration.url(),
+					_deepLTranslatorConfiguration.authKey(), entry.getValue(),
+					sourceLanguageCode, targetLanguageCode));
+		}
+
+		return new TranslatorPacket() {
+
+			@Override
+			public long getCompanyId() {
+				return translatorPacket.getCompanyId();
+			}
+
+			@Override
+			public Map<String, String> getFieldsMap() {
+				return translatedFieldsMap;
+			}
+
+			@Override
+			public String getSourceLanguageId() {
+				return translatorPacket.getSourceLanguageId();
+			}
+
+			@Override
+			public String getTargetLanguageId() {
+				return translatorPacket.getTargetLanguageId();
+			}
+
+		};
+	}
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_deepLTranslatorConfiguration = ConfigurableUtil.createConfigurable(
+			DeepLTranslatorConfiguration.class, properties);
+	}
+
+	private String _getLanguageCode(String languageId) {
+		List<String> list = Arrays.asList(
+			StringUtil.split(languageId, CharPool.UNDERLINE));
+
+		return StringUtil.toUpperCase(list.get(0));
+	}
+
+	private String _translate(
+			String url, String authKey, String text, String sourceLanguageCode,
+			String targetLanguageCode)
+		throws PortalException {
+
+		try {
+			if (Validator.isBlank(text)) {
+				return text;
+			}
+
+			List<Translation> translations =
+				_translationManager.getTranslateResponse(
+					authKey, text, sourceLanguageCode, targetLanguageCode, url);
+
+			Translation translation = translations.get(0);
+
+			return translation.getText();
+		}
+		catch (IOException ioException) {
+			throw new TranslatorException(
+				"DeepL translator returns original text. " +
+					ioException.getLocalizedMessage());
+		}
+	}
+
+	private boolean _verifyLanguage(
+		List<String> deepLLanguages, String compareLanguage) {
+
+		if (Collections.disjoint(
+				deepLLanguages, Arrays.asList(compareLanguage))) {
+
+			_log.error(
+				"DeepL does not support " + compareLanguage +
+					". Abort processing translation.");
+
+			return false;
+		}
+
+		return true;
+	}
```

Comment:
The "verify" name is too ambiguous. The naming should reflect what it is verifying; in this case, something like "_isLanguageSupported" seems more appropriate.